Bottom: 31eb422573cb62a97474c2ec21723ace4252314f
Top:    15df15bd33d06a2cb4a4f87359ab75861fb80880
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-03 23:23:46 +0100

Refresh of hid-gtk-provide-a-simple-progr

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 87a7dbd..9fbd128 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -750,6 +750,7 @@ static struct progress_dialog *
 make_progress_dialog (void)
 {
   struct progress_dialog *pd;
+  GtkWidget *alignment;
   GtkWidget *vbox;
 
   pd = g_new0 (struct progress_dialog, 1);
@@ -758,15 +759,23 @@ make_progress_dialog (void)
   gtk_window_set_title (GTK_WINDOW (pd->dialog), _("Progress"));
 
   pd->message = gtk_label_new (NULL);
+  gtk_label_set_width_chars (GTK_LABEL (pd->message), 30);
+  gtk_misc_set_alignment (GTK_MISC (pd->message), 0., 0.);
+
   pd->progress = gtk_progress_bar_new ();
+  gtk_widget_set_size_request (pd->progress, -1, 26);
 
-  vbox = gtk_vbox_new (false, 8);
+  vbox = gtk_vbox_new (false, 0);
   gtk_box_pack_start (GTK_BOX (vbox), pd->message, true, true, 8);
   gtk_box_pack_start (GTK_BOX (vbox), pd->progress, false, true, 8);
 
+  alignment = gtk_alignment_new (0., 0., 1., 1.);
+  gtk_alignment_set_padding (GTK_ALIGNMENT (alignment), 8, 8, 8, 8);
+  gtk_container_add (GTK_CONTAINER (alignment), vbox);
+
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (pd->dialog)->vbox),
-                      vbox, true, true, 0);
-  gtk_widget_show_all (vbox);
+                      alignment, true, true, 0);
+  gtk_widget_show_all (alignment);
 
   return pd;
 }
